sigma/backends/loki/loki.py

```diff
@@ -881,6 +908,23 @@ def convert_condition_and(
             )
         )
 
+    def convert_condition_field_eq_field(
+        self, cond: SigmaFieldReference, state: ConversionState
+    ):

```

Comment:
Could provide the output type here (`Union[str, DeferredQueryExpression]`)

Reply:
So I tried this, but the condition being joined with an `and` instead of a pipe `|`, which makes sense because the spec describes this as an and condition when

```
field|fieldref: fieldA
field2|fieldref: fieldB
```

Gets converted to

```
label_format match_0=`...` | match_0=`true` and label_format match_1=`...` | match_1=`true`
```

which isn't a valid LogQL so I need to get this to convert into

```
label_format match_0=`...` | match_0=`true` | label_format match_1=`...` | match_1=`true`
```

I think I'll update the `convert_condition_and` to check for field ref on either side and use a different joiner if that's the case

Reply:
A better approach might be to combine all of the `LogQLDeferredFieldRefExpression`s into a single expression that ends up containing:

```
| label_format match_0=... match_1=... | match_0=`true` and match_1=`true`
```